jigsawpy/libsaw.py

```diff
@@ -60,6 +60,13 @@
     if (not JLIBNAME.is_file()):
         JLIBNAME = Path()
 
+if not JLIBNAME.is_file():
+#---------------------------- search python environment for library
+    SYSLIB = {"Windows": "jigsaw.dll", "Linux": "libjigsaw.so", "Darwin": "libjigsaw.dylib"}
+    JLIBNAME = Path("/".join(sys.executable.split("/")[:-2])) / "lib" / SYSLIB[platform.system()]
+    if not JLIBNAME.is_file():
+        JLIBNAME = Path()
+
```

Comment:
@jreniel, do you mind explain this workflow a little --- somehow the `jigsaw` binaries are being installed inside `/usr/bin/python` or something along these lines? Or is the look-up for the `jigsaw` binaries just not working correctly?

Reply:
For most cases, Python "canonical" environments are created using `python -m venv my_pyenv`. When you do this, you can then activate that environment by `source my_pyenv/bin/activate` to override the current environment in favor of the `my_pyenv' environment. This simply looks for libjigsaw in `my_env/lib`. Using this example, `JLIBNAME` would be `my_env/lib/libjigsaw.*`.

Reply:
Here's a gist I wrote a while ago that briefly explains this pyenv business.

https://gist.github.com/jreniel/c2dd4f2f68f9d8172355461b5337f236